src/lib/getClasses.ts

```diff
@@ -14,15 +24,45 @@ const getClasses = (
   renderer: IRenderer,
   componentStyles: IComponentPartStylesInput,
   styleParam: ComponentStyleFunctionParam,
+  extra: any,
 ): IComponentPartClasses => {
   // root, icon, etc.
   const componentParts: string[] = Object.keys(componentStyles)
 
-  return componentParts.reduce((classes, partName) => {
-    classes[partName] = renderer.renderRule(callable(componentStyles[partName]), styleParam)
+  const { className: componentClassName } = extra
+
+  return componentParts.reduce(
+    (classes, partName) => {
+      // handle state section differently
+      // DISCLAIMER: dirty code alert!
+      if (partName.startsWith('on')) {
+        const pseudoStateName = partName.substr(2).toLowerCase()
```

Comment:
Totally supporting the work done here. In my opinion we can start to clear the code and move this further. As for the names of these parts, I would suggest adding the prefix (pseudo). I know this is dirty code as an example, but I think after renaming this it would be pretty much ready for review. What are your thoughts on this @kuzhelov ?

Reply:
absolutely agree :) still, one thing that I would like to test before making it 'PR ready' is performance - specifically, would like to estimate how costly it is for us to calculate these hashes, will try on the most contrived case and measure the time spent on that (and, essentially, how it compares to overall rerendering time)